core/services/head_tracker.go

```diff
@@ -133,22 +133,18 @@ type HeadTracker struct {
 	listenForNewHeadsWg   sync.WaitGroup
 	backfillMB            utils.Mailbox
 	subscriptionSucceeded chan struct{}
-	logger                *logger.Logger
 }
 
 // NewHeadTracker instantiates a new HeadTracker using the orm to persist new block numbers.
 // Can be passed in an optional sleeper object that will dictate how often
 // it tries to reconnect.
-func NewHeadTracker(store *strpkg.Store, callbacks []strpkg.HeadTrackable, sleepers ...utils.Sleeper) *HeadTracker {
+func NewHeadTracker(l *zap.SugaredLogger, store *strpkg.Store, callbacks []strpkg.HeadTrackable, sleepers ...utils.Sleeper) *HeadTracker {
```

Comment:
I think for consistency we'd want to pass logger.Logger in here right? Otherwise the behaviour will be different between the global and per service logger